dascore/proc/adaptive_spectral_filter.py

```
@@ -0,0 +1,487 @@
+        msg = (
+            f"adaptive spectral array filters require 1D or 2D input; got {data.ndim}D."
+        )
+        raise ValueError(msg)
+    if len(window_size) != data.ndim or len(overlap) != data.ndim:
+        msg = "window_size and overlap must match the input dimensionality."
+        raise ValueError(msg)
+    if not np.isfinite(exponent):
+        msg = "exponent must be finite."
+        raise ValueError(msg)
+
+    for axis, (window, axis_overlap) in enumerate(zip(window_size, overlap)):
```

Ruff (0.15.12)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)